sonar-batch/src/main/java/org/sonar/batch/bootstrap/ServerClient.java

```diff
@@ -102,31 +109,64 @@ public String request(String pathStartingWithSlash, String requestMethod, boolea
   }
 
   public String request(String pathStartingWithSlash, String requestMethod, boolean wrapHttpException, @Nullable Integer timeoutMillis) {
-    InputSupplier<InputStream> inputSupplier = doRequest(pathStartingWithSlash, requestMethod, timeoutMillis);
+    final byte[] buf = load(pathStartingWithSlash, requestMethod, wrapHttpException, timeoutMillis);
     try {
-      return IOUtils.toString(inputSupplier.getInput(), "UTF-8");
-    } catch (HttpDownloader.HttpException e) {
-      throw wrapHttpException ? handleHttpException(e) : e;
-    } catch (IOException e) {
+      return new String(buf, "UTF-8");
+    } catch (UnsupportedEncodingException e) {
       throw new IllegalStateException(String.format("Unable to request: %s", pathStartingWithSlash), e);
     }
   }
 
   public InputSupplier<InputStream> doRequest(String pathStartingWithSlash, String requestMethod, @Nullable Integer timeoutMillis) {
+    final byte[] buf = load(pathStartingWithSlash, requestMethod, false, timeoutMillis);
+
+    return new InputSupplier<InputStream>() {
+      @Override
+      public InputStream getInput() throws IOException {
+        return new ByteArrayInputStream(buf);
+      }
+    };
+  }
+
+  private byte[] load(String pathStartingWithSlash, String requestMethod, boolean wrapHttpException, @Nullable Integer timeoutMillis) {
     Preconditions.checkArgument(pathStartingWithSlash.startsWith("/"), "Path must start with slash /");
     String path = StringEscapeUtils.escapeHtml(pathStartingWithSlash);
-
     URI uri = URI.create(getURL() + path);
+
     try {
+      if(GET.equals(requestMethod) && mode.isPreview()) {
+        return cache.get(uri.toString(), new HttpValueLoader(uri, requestMethod, timeoutMillis));
+      }
+      else {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Move this "else" on the same line that the previous closing curly brace. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3ARightCurlyBraceSameLineAsNextBlockCheck)